src/main/java/com/ldtteam/structurize/client/gui/WindowShapeTool.java

```diff
@@ -541,21 +668,27 @@ private void rotateLeftClicked()
     {
         rotation = (rotation + ROTATE_THREE_TIMES) % POSSIBLE_ROTATIONS;
         updateRotation(rotation);
+        updateRotationState();
     }
 
 
     /*
      * ---------------- Miscellaneous ----------------
      */
 
+    private void clearAndClose()
+    {
+        Settings.instance.resetBlueprint();
+        Network.getNetwork().sendToServer(new LSStructureDisplayerMessage(null, false));
+        close();
+    }
+
     /**
      * Cancel the current structure.
      */
     private void cancelClicked()
     {
-        Settings.instance.reset();
-        Network.getNetwork().sendToServer(new LSStructureDisplayerMessage(null, false));
-        close();
+        clearAndClose();
```

Comment:
no need for alias method

Reply:
There is, because it's called from more than one place.  True, the cancel handler *could* call that directly but that is poor style.

Reply:
Like you literally extracted content of this method to new method which does exactly the same

Reply:
It doesn't, actually.  I wrote a new method that does something slightly different, and then called it from three places.  One of those places happens to be the cancel button click handler, which also happens to not do anything else, but that isn't a good reason to call a utility method directly as the cancel button click handler instead of actually having a named cancel button click handler.  It's possible that in the future some other action may be needed on cancel that doesn't happen for the other cases.

Reply:
Having an alias has no good reason either, also you named method updateRotationState and call it from mirror method - how does that make sense I don't know, and future code can arise in day as well as ten years. simply having alias method has no valid reason in my eyes, clearAndClose can surely be target method for button handler

Reply:
No, it can't.  That's just silly.